File: split/slowprint.pas
procedure slowprint(s: integer);
var
  j: poolpointer;
begin
  if (s >= strptr) or (s < 256) then
    print(s)
  else
  begin
    j := strstart[s];
    while j < strstart[s + 1] do
    begin
      print(strpool[j]);
      j := j + 1;
    end;
  end;
end;

File: cpp/slowprint.cpp
#include "slowprint.h"
#include "printchar.h"


void slowprint(int s)
{
	if (s >= strptr || s < 256)
		printchar(s);
	else
		for (int j = strstart[s]; j < strstart[s+1]; j++)
			printchar(strpool[j]);
}

void slowprint(const std::string &s)
{
	for (auto c: s)
		printchar(c);
}
